### ISSUE TITLE
Bitwise operators on `ndarray.at`

### ISSUE THREAD

**Issue by JeppeKlitgaard** [google/jax#15601]
I have run into a use-case where it would be quite convenient to use the bitwise operators (OR, AND, XOR, NOT) with the `Array.at` methods.

I imagine they simply haven't been implemented because JAX is relatively rarely used with masks and boolean data.

**Comment by jakevdp** [comment]
Indeed these haven't been implemented. In theory you should be able to use the `apply()` method for this, but it's not well-tested and seems to be broken for this case:
```python
x = jnp.arange(4)
y = x.at[1:].apply(lambda x: x & 1)
print(y)  # this looks wrong...
[0 1 2 3]
```
I think there's a long-open bug for that somewhere that's assigned to me /ducks

Another option would be to use `where` statements, e.g. something like this:
```python
y = jnp.where(jnp.arange(len(x)) >= 1, x & 1, x)
print(y)
# [0 1 0 1]
```

**Comment by JeppeKlitgaard** [comment]
Thanks! I ended up going for `at[...].set(...)` where the argument to set can then be computed using regular slicing and the `|, &, ^` operators. 

Would be neat to have these added, as they come up when using masking arrays a fair bit.

**Comment by jakevdp** [comment]
After #15643 is merged, you should be able to do updates for any function using this pattern:
```python
y = x.at[1:].apply(lambda x: x & 1)
```

**Comment by jakevdp** [comment]
I'm going to close this issue; I think it's unlikely that `Array.at` will grow specific methods for bitwise and other binary operations, and `apply` provides a good solution for this case.

**Comment by SobhanMP** [comment]
Would it be possible to overload 'mul' to mean 'and' and 'add' to 'or'?

**Comment by SobhanMP** [comment]
I later realized that max and min already do this.
